router/auth.go

```diff
@@ -136,6 +137,15 @@ func authInjectYellowfin(log *log.Logger, w http.ResponseWriter, req *http.Reque
 		return false
 	}
 
+	// Used to configure the module filter used in YF logins
+	if req.URL.Path == "/yellowfin/setModuleFilter" {
+		var filterValue []byte
+		filterValue, _ = ioutil.ReadAll(req.Body)
+		target.filterMap[authData.Identity] = string(filterValue) // set this user's filter
+		delete(target.tokenMap, authData.Identity)                // invalidate session (YF relogin)
```

Comment:
You need to be holding target.lock here. Same as the code below. It looks to me it doesn't make sense to add a new mutex for filterMap - just use the existing 'lock' to guard everything, including the new 'filterMap' data member.